### PR TITLE
PLAT-61277: Update i18n docs

### PR DESCRIPTION
**Pull request by viodragon2**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Update `i18n` documentation.

A couple of things I noticed in our current doc (http://enactjs.com/docs/modules/i18n/$L/):
1. `util` module is not shown. I'm assuming this module is public, here. Though our recommendation is to use `i18nDecorator` and `Uppercase` HOC over using `isRTLText` or all the casing functions, it seems reasonable to make it public IMO. I'm open for debate on this.
2. `docs/iLib` and `docs/updateLocale` link is hard to find
3. There are mentions of `moonstone` in `docs/index.md` and `update-locale.md`. I wonder if these need to be modified so that it shows the example of using in non-moonstone context.

Enact-DCO-1.0-Signed-off-by: Stephen Choi <stephen.choi@lge.com>

### CODE REVIEW

File: packages/i18n/I18nDecorator/I18nDecorator.js [hunk 1]
```diff
@@ -52,7 +53,22 @@ const IntlHoc = hoc((config, Wrapped) => {
...
+			/**
+			 * Classname for a root app element.
+			 *
+			 * @type {String}
+			 * @public
+			 */
 			className: PropTypes.string,
+
+			/**
+			 * Locale string in the format.
```
Comment (webOS101) [inline packages/i18n/I18nDecorator/I18nDecorator.js]: In what format?

File: packages/i18n/I18nDecorator/getI18nClasses.js [hunk 2]
```diff
@@ -1,14 +1,21 @@
 import {isNonLatinLocale, isRtlLocale} from '../locale';
 import LocaleInfo from '../ilib/lib/LocaleInfo';
 
+/**
```
Comment (webOS101) [inline packages/i18n/I18nDecorator/getI18nClasses.js]: This doesn't need a jsdoc block, it's not just private, it's not exported from the module at all.

File: packages/i18n/Uppercase/Uppercase.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 /**
- * Exports the {@link i18n/Uppercase.Uppercase} Higher-Order Component (HOC)
+ * Provide higher-order component interface for handling locale-aware uppercasing.
```
Comment (webOS101) [inline packages/i18n/Uppercase/Uppercase.js]: Provides

File: packages/i18n/$L/$L.js [hunk 4]
```diff
@@ -1,22 +1,24 @@
 /**
- * Exports the {@link i18n/$L.$L} function and {@link i18n/$L.toIString} function to map to
- * translated strings.
+ * Provides methods to map to transtalted strings.
```
Comment (webOS101) [inline packages/i18n/$L/$L.js]: typo

File: packages/ui/resolution/ResolutionDecorator.js [hunk 5]
```diff
@@ -104,7 +104,8 @@ const ResolutionDecorator = hoc(defaultConfig, (config, Wrapped) => {
 		 * Compare our current version of the resolved resolution class names with a fresh
 		 * initialization of RI.
 		 *
-		 * @return {String|undefined} A string of new class names or undefined when there is no change.
+		 * @returns {String|undefined} A string of new class names or undefined when there is no change.
+		 * @private
```
Comment (webOS101) [inline packages/ui/resolution/ResolutionDecorator.js]: This, too, shouldn't be in a jsdoc block, it's not a public method.

**Comment by Djspaceg** [comment]
Alphabetized the exports of i18n/util/util.js

**Pull request merged**